### PR TITLE
Arc signatures

### PR DESCRIPTION
**Pull request by jrlevine**
Patches to support ARC seals in addition to DKIM signatures

Notes:

- ARC config is similar to DKIM config. If you have DKIM working, and enable arc_feature in robots.txt and set arc_srvid to the srvid in your MTA's Authentication-Results, it should just work
- ARC uses the same kind of keys as DKIM and defaults to the same domain, selector, and keyfile as DKIM, which is usually correct
- ARC will only sign a message if Sympa also adds a DKIM signature. Set Sympa to DKIM sign everything, since it can only help deliverability
- ARC needs an Authentication-Results header applied by the incoming MTA. If your MTA doesn't already do that, there are drop-in milters
- ARC only applies to forwarded messages, so there are no ARC seals on  Sympa generated mail such as welcome messages and digests (but do DKIM sign them)

### CODE REVIEW

**Comment by ikedas** [comment]
Related to #153.

File: src/lib/Sympa/ListDef.pm [hunk 1]
```diff
@@ -2153,8 +2153,75 @@ our %pinfo = (
...
+                    'The ARC "d=" tag, is the domain of the sealing entity. The list domain MUST be included in the "d=" domain',
+                'format'     => '\S+',
+                'occurrence' => '0-1',
+                'default'    => {'conf' => 'arc_signer_domain'}
+            },
+        },
+        'occurrence' => '0-1'
+    },
+
+    'dkim_signature_apply_on' => {
```
Comment (ikedas) [inline src/lib/Sympa/ListDef.pm]: dkim_signature_apply_on is duplicated.

File: default/edit_list.conf [hunk 2]
```diff
@@ -140,6 +140,9 @@ dkim_parameters			owner,privileged_owner 		hidden
 dkim_signature_apply_on		listmaster	 		write
 dkim_signature_apply_on		owner,privileged_owner 		hidden
 
+arc_parameters			listmaster			write
+arc_parameters			owner,privileged_owner		hidden
+
```
Comment (ikedas) [inline default/edit_list.conf]: arc_feature would be better to be added.

File: src/lib/Sympa/ListDef.pm (hunk 1 above)
Comment (jrlevine) [reply src/lib/Sympa/ListDef.pm]: Correct, cut and paste error. Just delete the second one.

File: default/edit_list.conf (hunk 2 above)
Comment (jrlevine) [reply default/edit_list.conf]: Yes, added

**Comment by ikedas** [comment]
I found no particluar problems anymore with review on desktop. Let's merge this pr and look it works fine.

**Pull request merged**